### PR TITLE
pkg5: Provide a boolean switch to enable/disable publisher refresh

### LINKED ISSUE #46682: Solaris - pkg5 module - Provide a boolean switch to enable/disable publisher refresh
##### SUMMARY
The pkg5 module used for Solaris installs is missing an option to disable refreshing publishers. ("--no-refresh").

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
pkg5

##### ADDITIONAL INFORMATION
 At the moment I can't update or install packages on a Solaris Vagrant box because Solaris 11.4 is released and my system wants to update to 11.4 packages. The problem is it can't update because it's an unsupported Solaris and thus doesn't get a required update that should be downloaded before migrating to Solaris 11.4.

I want to be able to prevent publisher package refresh so that I don't run into this problem.

```yaml
    - name: install mutt MTA
      become: true
      pkg5:
        name: mail/mutt
        state: present
        refresh: false
```

### PR DESCRIPTION
**Pull request by jpdasma**
##### SUMMARY
Provide a boolean switch to enable/disable publisher refresh for Solaris IPS

Fixes #46682

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
`pkg5.py`

##### ANSIBLE VERSION
`2.8`

### CODE REVIEW

**Review by bcoca — APPROVED** [review]

File: lib/ansible/modules/packaging/os/pkg5.py [hunk 1]
```diff
@@ -135,9 +147,14 @@ def ensure(module, state, packages, params):
     else:
         beadm = []
 
+    if not params['refresh']:
```
Comment (mavit) [inline lib/ansible/modules/packaging/os/pkg5.py]: Perhaps I'm being picky, but would this be slightly easier to read if the `not` was removed and the order of the following blocks was swapped?

Comment (jpdasma) [reply lib/ansible/modules/packaging/os/pkg5.py]: yeah I could do that. I initially did it that way but forgot why I reverted the logic. I think I reverted it due to the default behavior of pkg5, now that I think about it doesn't make sense. I'll revert it later.

**Review by dagwieers — APPROVED** [review]
LGTM

**Comment by dagwieers** [comment]
rebuild_merge

File: lib/ansible/modules/packaging/os/pkg5.py [hunk 2]
```diff
@@ -74,6 +85,7 @@ def main():
             state=dict(type='str', default='present', choices=['absent', 'installed', 'latest', 'present', 'removed', 'uninstalled']),
             accept_licenses=dict(type='bool', default=False, aliases=['accept', 'accept_licences']),
             be_name=dict(type='str'),
+            refresh=dict(type='bool', default=True),
```
Comment (danowar2k) [inline lib/ansible/modules/packaging/os/pkg5.py:88]: Should there be a comma at the end? Does Python ignore this?

Comment (mavit) [reply lib/ansible/modules/packaging/os/pkg5.py:88]: @danowar2k: https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas

**Pull request merged**